.github/workflows/go-mod-cache.yml

```diff
@@ -0,0 +1,53 @@
+name: Go Module Cache
```

Comment:
Since we're also using this on /chainlink, we should we move this to the /.github repo?

Reply:
That's a good idea. It would require us consolidating our `setup-go` actions, because the one here, and in /chainlink, and in .github have different behaviours. Two of which are tailored to the repo, and the setup-golang in .github is very general.

I'll create a backlog task.